Sources/SourceControl/RepositoryManager.swift

```diff
@@ -325,12 +325,21 @@ public class RepositoryManager: Cancellable {
                     }
                 }
             } catch {
-                cacheUsed = false
-                // Fetch without populating the cache in the case of an error.
-                observabilityScope.emit(warning: "skipping cache due to an error: \(error)")
-                // it is possible that we already created the directory from failed attempts, so clear leftover data if present.
-                try? self.fileSystem.removeFileTree(repositoryPath)
-                try self.provider.fetch(repository: handle.repository, to: repositoryPath, progressHandler: updateFetchProgress(progress:))
+                // If we are offline and have a valid cached repository, use the cache anyway.
+                if isOffline(error) && self.provider.isValidDirectory(cachedRepositoryPath) {
+                    observabilityScope.emit(warning: "no connectivity, using previously cached repository state")
```

Comment:
probably "info" instead of "warning"

Reply:
Looks like this now in verbose mode now:

```
Fetching https://github.com/apple/swift-package-manager from cache
warning: no connectivity, using previously cached repository state
info: using previously cached repository state for swift-package-manager
Fetched https://github.com/apple/swift-package-manager (0.05s)
Fetching https://github.com/jpsim/yams from cache
info: using previously cached repository state for yams
Fetched https://github.com/jpsim/yams (0.05s)
```

In non-verbose, we only get the singular warning, of course.